### PR TITLE
closes #1012 - allow string[] to be used in an @Extension

### LINKED ISSUE #1012: Declaration of ExtensionType does not fit for Amazon API Gateway
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `@tsoa/runtime@3.8.0` for the project I'm working on.

We are using TSOA to generate the OpenAPI spec which we feed, via Serverless Framework, into Amazon API Gateway to set up our API. In order to make that work, we use `@Extension` declarations like this:

```javascript
@Extension("x-amazon-apigateway-integration", {
        "uri": {
            "Fn::Join": [
                "",
                [
                    "arn:aws:apigateway:${env:region}:lambda:path/2015-03-31/functions/",
                    "arn:aws:lambda:${env:region}:${env:AccountId}:function:MyLambdaFunctionName",
                    "/invocations"
                ]
            ]
        },
        "passthroughBehavior": "when_no_match",
        "credentials": {
            "Fn::GetAtt": [
                "MyRestApiRole",
                "Arn"
            ]
        },
        "httpMethod": "POST",
        "type": "aws_proxy"
    })
```

However, we needed to add `string[]` to the declaration of `ExtensionType` in TSOA, to make it work. Until now, we created patches for every TSOA update. However, there might be some other folks out there doing similar stuff as we do. So, do you mind extending the declaration of `ExtensionType`?

Kind regards
Oliver

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/@tsoa/runtime/dist/decorators/extension.d.ts b/node_modules/@tsoa/runtime/dist/decorators/extension.d.ts
index d125ad6..5d1d030 100644
--- a/node_modules/@tsoa/runtime/dist/decorators/extension.d.ts
+++ b/node_modules/@tsoa/runtime/dist/decorators/extension.d.ts
@@ -1,6 +1,7 @@
 export declare function Extension(_name: string, _value: ExtensionType | ExtensionType[]): Function;
 export declare type ExtensionType =
   | string
+  | string[]
   | {
       [name: string]: ExtensionType | ExtensionType[];
     };
diff --git a/node_modules/@tsoa/runtime/src/decorators/extension.ts b/node_modules/@tsoa/runtime/src/decorators/extension.ts
index 2ae0ec2..98a1d8a 100644
--- a/node_modules/@tsoa/runtime/src/decorators/extension.ts
+++ b/node_modules/@tsoa/runtime/src/decorators/extension.ts
@@ -4,4 +4,4 @@ export function Extension(_name: string, _value: ExtensionType | ExtensionType[]
   };
 }
 
-export type ExtensionType = string | { [name: string]: ExtensionType | ExtensionType[] };
+export type ExtensionType = string | string[] | { [name: string]: ExtensionType | ExtensionType[] };
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

### PR DESCRIPTION
**Pull request by oliver-brm**
### All Submissions:

- [✔️] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
- [✔️] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
- [✔️] Have you written unit tests?
- [❌] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)? (Already existed as `InvalidExtensionControllerGenerator` test suite)
- [✔️] This PR is associated with an existing issue?

**Closing issues**

Closes #1012.

**Test plan**

The added test suite `ValidExtensionControllerGenerator` makes sure that `@Extension` declarations, similar to those we need to feed into Amazon API Gateway, do work. It's somewhat a double-check because without allowing `string[]` in `@ExtensionType`, TypeScript would not compile at all.

### CODE REVIEW

File: tests/unit/swagger/definitionsGeneration/metadata.spec.ts [hunk 1]
```diff
@@ -24,6 +24,14 @@ describe('Metadata generation', () => {
     });
   });
 
+  describe('ValidExtensionControllerGenerator', () => {
```
Comment (WoH) [inline tests/unit/swagger/definitionsGeneration/metadata.spec.ts:27]: We already have specs for existing extensions for both v2 and v3 of OpenAPI.
Can you please extend these? Not throwing is too weak of an assertion.

Refs:
- https://github.com/lukeautry/tsoa/blob/master/tests/fixtures/controllers/methodController.ts#L158
- https://github.com/lukeautry/tsoa/blob/master/tests/unit/swagger/schemaDetails.spec.ts#L495
- https://github.com/lukeautry/tsoa/blob/master/tests/unit/swagger/schemaDetails3.spec.ts#L2144

**Comment by oliver-brm** [comment]
@WoH OK, I added the fixture/tests to `methodController.ts`, `schemaDetails.spec.ts`, `schemaDetails3.spec.ts` and `metadata.spec.ts`.

File: tests/unit/swagger/definitionsGeneration/metadata.spec.ts (hunk 1 above)
Comment (oliver-brm) [reply tests/unit/swagger/definitionsGeneration/metadata.spec.ts]: @WoH OK, I added the fixture/tests to methodController.ts, schemaDetails.spec.ts, schemaDetails3.spec.ts and metadata.spec.ts.

**Comment by WoH** [comment]
LGTM, thanks for submitting a PR! 🚀

**Pull request merged**